myning/utils/race_rarity.py

```diff
@@ -38,8 +38,33 @@
 RACE_WEIGHTS = [150, 75, 40, 20, 10, 7, 4]
 
 
-def get_recruit_species(highest_rarity: int):
+def _recruit_in_tier(tier: list[CharacterRaces]) -> CharacterRaces:
+    if len(tier) == 0:
+        raise ValueError("Empty tier")
     player = Player()
+    facility = ResearchFacility()
+    undiscovered = [
+        RACES[race_index] for race_index in tier if RACES[race_index] not in player.discovered_races
+    ]
+    undiscovered_probability = len(undiscovered) / len(tier)
+    percent_increase = 0
+    if facility.has_research("species_discovery"):
+        percent_increase = RESEARCH["species_discovery"].player_value / 100
+    undiscovered_probability += undiscovered_probability * percent_increase
+    discovered_probability = 1 - undiscovered_probability
+    undiscovered_weight = undiscovered_probability / len(undiscovered)
+    discovered_weight = discovered_probability / (len(tier) - len(undiscovered))
```

Comment:
I don't understand these two lines

Reply:
These will be the final weights in random.choices for undiscovered races, so if you have no boosts, and there are 5 species in a tier, it will be 1 / 5 for each.

Reply:
These will be fractions. I bet choices understands those, but I think I preferred having them all as ints